src/groupByKey.ts

```diff
@@ -2,6 +2,18 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const gropedObj: GroupsMap<T> = {};
```

Comment:
There seems to be a typo in the variable name `gropedObj`. It should be `groupedObj` to accurately reflect its purpose and to avoid confusion.